### DISCUSSION TITLE
DBeaver is the best database IDE and has the sludgiest graphical performance

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#19162] [General]
I love DBeaver. I've found it's more reliable or user-friendly or powerful than other database IDEs I've used in the last several years.

However, it's a slog waiting for things to animate.

I'm on a 2019 Mac with 16GB of RAM, and have used DBeaver on every macOS major version from Catalina to Ventura, and it has been by far and most-consistently the sludgiest-behaving app on it, and doesn't even compare to the other IDE's performance. I've seen it behave similarly relatively slow on other computers, even newer faster ones, even Windows.

I don't understand the neglect of performance and its optimization in the app. Other apps either don't have an issue with this or have mastered it. Also, the app appears to be slowing down slightly over time and updates. As much as I love all the powerful features, it seems the app is over-analyzing too many of its own powerful features that it doesn't need to when in use.

I've tried it on other computers that also run fast with other IDEs (even non-database IDEs, with varying complexity, some very complex), and it's still by far the sludgiest. It's honestly turning others off to it that I'm trying to convert (from an outdated/under-featured app), since the app they're used to just runs as fast as any other app.

How do we get the DBeaver developers to prioritize UI/UX interactivity/animation speed and performance optimization? There's already so many features that I don't know that I'll ever learn them all, which is beyond great, but the app has to perform well to convert people from the rut of their fast archaic or under-featured apps.

**Comment by serge-rider** [comment]
@givihuda Thank you for the feedback.

Actually we are working on the UI/UX performance all the time. We know about performance issues on MacOS with HighDPI resolution, especially when you maximize DBeaver window and there is a lot of info on a screen. 
Basically it happens because of UI rendering issues in Eclipse platform on MacOS specifically. But we trying to improve it anyway.

Coiuld you describe most typical of your usecases when UI become laggy/freezes?

**Reply by givihuda** [reply]
That's good to know you're working on UI/UX. I use DBeaver maximized on a QHD screen. I mention the performance issues because it runs much slower than other apps I use. DBeaver makes sense with a lot of info on the screen, because as a database IDE I want to see a lot of things and manage a lot of things there. Even multiple instances of multiple tabs and panels for different apps or databases or tasks. I do that in JetBrains code IDEs almost every day of the week, and often with multiple projects open with their own panels and tabs each. It runs very fast. I don't understand why DBeaver, doing basically the same things, can't also run fast. So I don't think having a lot of info on the screen is any sort of excuse for poor performance.

Maybe the issue is Eclipse? I haven't run Eclipse in a while, but I don't remember it having performance issues like this. Maybe I should try that again? Should I try only its database navigator? Is that an adequate comparison to check for performance?

It's possible that DBeaver developers think they're optimizing or getting great performance when they're not aware of better ways. I can't advise on that, but there are architecture patterns or things to look for to minimize inefficiencies. My guess is there are just inefficiencies and bottlenecks in many places in the app. Maybe it would be worth finding an optimization consultant for the type of code stack DBeaver uses.

Right now my typical usage of DBeaver is probably much simpler than you think. I start with the Database Navigator panel, expand a database connection, a database schema, and its Tables list. Sometimes I need to expand a table and its columns, sometimes I need to right-click a table and view its diagram or data (to run a simple filter search), sometimes I need to create a new SQL script to type in manual SQL... I can have between 1 tab and 20 open, sometimes split between panes. Performance issues seem to typically be there though. Just expanding things is slower than in other apps. Shifting the window size is slower. Moving tabs is slower. Clicking into things is slower. Moving the window is slower. Everything is slower than in other apps I use on the Mac. I can't blame myself or the Mac when DBeaver is the only thing running slow. That's why I think DBeaver has some inefficiencies that are creating bottlenecks that are causing performance issues that can be fixed through optimization.

**Reply by givihuda** [reply]
Oh, it doesn't seem to behave much better on an FHD screen.

Also, it often crashes if I try to open a tab in its own window. That's probably a separate issue, but who knows, it may be related to problematic coding architecture, whether that's due to DBeaver coders or Eclipse coders, who knows.

**Reply by E1izabeth** [reply]
Thank you for feedback.
